GEECS-PythonAPI/geecs_python_api/controls/data_acquisition/scan_manager.py

```diff
@@ -36,15 +37,15 @@ class ScanManager():
     the desired saving configuration. 
     """
     
-    def __init__(self, experiment_dir=None, device_manager=None, data_interface=None):
-        self.device_manager = device_manager or DeviceManager(experiment_dir=experiment_dir)
-        self.data_interface = data_interface or DataInterface()
+    def __init__(self, experiment_dir=None, shot_control_device=""):
+        self.device_manager = DeviceManager(experiment_dir=experiment_dir)
+        self.data_interface = DataInterface()
         self.action_manager = ActionManager(experiment_dir=experiment_dir)
         self.data_logger = DataLogger(experiment_dir, self.device_manager)  # Initialize DataLogger
 
         self.save_data = True
 
-        self.shot_control = GeecsDevice('U_DG645_ShotControl')
+        self.shot_control = GeecsDevice(shot_control_device)
```

Comment:
I get the intention of generalizing this beyond HTU. I suspect the shot_control_device is being read from the database or something? The downstream implementation is still hardcoded to use a specific channel for our DG645...I guess my point is, this generalization is only half of the story...Probably a good step in the right direction though

Reply:
Just looked a bit more. Looks like the shot_control_device is purely user defined from the GUI? A shot control device is generally (though not "required") as part of setting up any GEECS device and can be read from the database. In any case, probably something to think about later

Reply:
Right, although I also have this as something that is stored within the .config file for the GEECS Python API.  So in general this is something that *could* be set in the GUI but should instead just be set in the config file and you won't have to think about it.

Now that I am thinking about it, something that could be good eventually is something on this GUI that automatically generates a .config file with appropriate elements if it doesn't exist.  I'll put that on the wishlist, but for now I may just provide a .config file in this folder to use for HTU.